shared-libs/message-utils/src/index.js

```diff
@@ -96,21 +96,37 @@ const applyPhoneFilters = function(config, phone) {
   return phone;
 };
 
+const normalizeRecipient= function(recipient) {
+  const toArray = Array.isArray(recipient) ? recipient : [recipient];  
+  return toArray.map(
+    r => (r === null ? '': String(r).trim())
+  ).filter(r => r.length > 0);
+};
+
 const getRecipient = function(context, recipient, defaultToSender = true) {
   if (!context) {
     return;
   }
 
   const from = context.from || context.contact?.phone;
-  recipient = recipient?.trim();
+  recipient = normalizeRecipient(recipient);
 
-  if (!recipient) {
+  if (!recipient.length) {
     return from;
   }
 
-  const phone = resolveRecipient(context, recipient);
+  const phone = resolveMany(context, recipient);
+  return phone || (defaultToSender && from) || recipient[0];
+};
 
-  return phone || (defaultToSender && from) || recipient;
+const resolveMany = (context, recipients) => {
+  for (const recipient of recipients) {
+    const phone = resolveRecipient(context, recipient);
+    if (phone) {
+      return phone;
+    }    
+  }
+  return null;
```

Comment:
Here the `null` return is not required. Omitting it will return `undefined` which will work fine in your code.